pkg/spoke/controllers/statuscontroller/availablestatus_controller.go

```diff
@@ -103,12 +105,23 @@ func (c *AvailableStatusController) syncManifestWork(ctx context.Context, origin
 	needStatusUpdate := false
 	// handle status condition of manifests
 	for index, manifest := range manifestWork.Status.ResourceStatus.Manifests {
```

Comment:
nit: `manifestWork.Status.ResourceStatus.Manifests` is an array of value (not references), consider avoid direct usage of `manifest` in case of racy condition

Reply:
good point. will add a todo here. I think it is the issue of this whole controller on potential race.